### ISSUE TITLE
Support for perlre's \K for group matching (aka "Zoom" anchors)

### ISSUE THREAD

**Issue by kenorb** [rust-lang/regex#462]
I'm looking for the equivalent sequence of [`\K`in perlre](https://perldoc.perl.org/perlre.html#Extended-Patterns). Here is the example using GNU `grep`:

```
$ printf "foo bar buzz" | ggrep -oP "foo \Kbar"
bar
```

> `\K` - Keep the stuff left of the \K

Related post: [Can grep output only specified groupings that match?](https://unix.stackexchange.com/q/13466/21471)

This is similar to ["Zoom" anchors Vim-like syntax](https://vi.stackexchange.com/q/3036/467).

I've tried with `rg`, but it failed with:

```
$ printf "foo bar buzz" | rg -o "foo \Kbar"
Error parsing regex near 'foo \Kbar' at character offset 5: Unrecognized escape sequence: '\K'.
(Hint: Try the --fixed-strings flag to search for a literal string.)
```

Same with `rg -o "foo (?<=bar)"` or `rg -o "foo (?=bar)"`.

Is there any similar syntax already, or it's a feature?

Basically, I need to grep string with `-o` and return only specific part of it.

**Comment by BurntSushi** [comment]
From the README:

> A Rust library for parsing, compiling, and executing regular expressions. Its syntax is similar to Perl-style regular expressions, but lacks a few features like look around and backreferences. In exchange, all searches execute in linear time with respect to the size of the regular expression and search text.

**Comment by ethanpailes** [comment]
@kenorb 

Does

    echo "foo bar bazz" | rg "^.*foo (bar).*$" --replace '$1'

do what you want? It seems like \K is just sugar for capture groups, or am I missing something?

**Comment by kenorb** [comment]
@ethanpailes Yep, this is what I was looking for, thank you!
